webui/react/src/stores/userSettings.tsx

```diff
@@ -141,8 +145,42 @@ class UserSettingsStore extends PollingStore {
     }
   }
 
+  /**
+   * This updates the value of a setting and persists it for future sessions.
+   * @param type The type of the value or an encoder of the value to JSON.
+   * @param key Unique key to store and retrieve the settings
+   * @param fn Function to update the value of the setting at `key`
+   */
+  public update<T extends t.Props, U = t.TypeOfProps<T>>(
+    type: t.TypeC<T>,
+    key: string,
+    fn: (value: U | undefined) => U,
+  ): void;
+  public update<T>(type: t.Type<T, Json>, key: string, fn: (value: T | undefined) => T): void {
+    this.#settings.update((settings) => {
+      return Loadable.map(settings, (settings) => {
+        return settings.update(key, (jsonValue) => {
+          let value: T | undefined = undefined;
+          if (jsonValue !== undefined) {
+            const attempt = type.decode(jsonValue);
+            if (isRight(attempt)) {
+              value = attempt.right;
+            }
```

Comment:
Should we log the error in case `isLeft` happens?

Reply:
I don't think we should display anything to the user since it's not the result of any of their actions but we could log out to console. What do you think?

Reply:
Logging to console sounds good, in case any thing happens, we don't have to hunt down the error

Reply:
Done.